packages/core/src/model.ts

```diff
@@ -89,16 +98,27 @@ export function createModel<
   creators: TModelCreators
 ): Model<
   TContext,
-  Cast<
-    EventFromEventCreators<Prop<TFinalModelCreators, 'events'>>,
-    EventObject
-  >,
+  Prop<TFinalModelCreators, 'events'> extends never
```

Comment:
this could potentially read a little bit better, wdyt?

```ts
'events' extends keyof TFinalModelCreators
```

Reply:
This didn't work when I tried it